lifelines/fitters/__init__.py

```diff
@@ -3009,3 +2965,142 @@ def predict_cumulative_hazard(self, df, ancillary_df=None, times=None, condition
                 index=times,
                 columns=df.index,
             )
+
+
+class Printer:
+    def __init__(
+        self, headers: List[Tuple], model: Type[BaseFitter], justify: Callable, decimals: int, header_kwargs: Dict
+    ):
+        self.headers = headers
+        self.model = model
+        self.decimals = decimals
+        self.justify = justify
+
+        for tuple_ in header_kwargs.items():
+            self.headers.append(tuple_)
+
+    def print(self):
```

Comment:
cc @pzivich this is how I implemented a HTML vs ASCII printer class. The code is pretty ugly ATM, but you can at least see the control flow.